kamailio/dispatcher-role.cfg

```diff
@@ -171,27 +172,35 @@ route[DISPATCHER_NEXT_ROUTE]
 
     if ($sht(failover=>$ci::counts) == $null) {
         $var(i) = 0;
+        $(avp(ds_dst)[$var(i)]) = $null; # Removing already failed dispatcher route
+
+        # Also try backup servers
+        if ($avp(ds_group) == 1 && ds_select_dst("2", "0")) {
+            xlog("L_CRIT", "$ci|log|Trying deliver call via backup dispatcher servers\n");
+        }
+
         while($(avp(ds_dst)[$var(i)]) != $null) {
             $sht(failover=>$ci[$var(i)]) = $(avp(ds_dst)[$var(i)]);
             $var(i) = $var(i) + 1;
-            if ($var(i) >= 3) {
+            if ($var(i) >= 2) {
                 break;
             }
         }
+
         $sht(failover=>$ci::counts) = $var(i);
         $var(failover_count) = $var(i);
     }
 
     # try to find a new media server to send the call to
-    if ($var(failover_count) > 1) {
+    if ($var(failover_count) > 0) {
         $var(failover_count) = $var(failover_count) - 1;
```

Comment:
Later need to change failover media servers order from `0` to 'MAX_FAILOVER_ATTEMPTS`
In this case backup servers tried first. Now primary servers tied first if `0 < backup_server < MAX_FAILOVER_ATTEMPTS`